src/ui/public/timefilter/lib/diff_time_picker_vals.js

```diff
@@ -18,28 +18,24 @@
  */
 
 import _ from 'lodash';
-import angular from 'angular';
 
-export function UtilsDiffTimePickerValsProvider() {
+const valueOf = function (o) {
+  if (o) return o.valueOf();
+};
 
-  const valueOf = function (o) {
-    if (o) return o.valueOf();
-  };
-
-  return function (rangeA, rangeB) {
-    if (_.isObject(rangeA) && _.isObject(rangeB)) {
-      if (
-        valueOf(rangeA.to) !== valueOf(rangeB.to)
-        || valueOf(rangeA.from) !== valueOf(rangeB.from)
-        || valueOf(rangeA.value) !== valueOf(rangeB.value)
-        || valueOf(rangeA.pause) !== valueOf(rangeB.pause)
-      ) {
-        return true;
-      }
-    } else {
-      return !angular.equals(rangeA, rangeB);
+export function areTimePickerValsDifferent(rangeA, rangeB) {
+  if (_.isObject(rangeA) && _.isObject(rangeB)) {
+    if (
+      valueOf(rangeA.to) !== valueOf(rangeB.to)
+      || valueOf(rangeA.from) !== valueOf(rangeB.from)
+      || valueOf(rangeA.value) !== valueOf(rangeB.value)
+      || valueOf(rangeA.pause) !== valueOf(rangeB.pause)
+    ) {
+      return true;
     }
+  } else {
+    return !_.isEqual(rangeA, rangeB);
```

Comment:
This technically is different than the old code (I think anyway) because angular.equals doesn't care about variables on an object that start with `$`.  But playing around with it, it doesn't seem to ever matter, so I think it's okay to change this.  It actually seems pretty weird to ever fall in here anyway - only seems to be the case if either rangeA or rangeB are undefined.